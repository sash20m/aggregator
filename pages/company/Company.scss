.layout {
    display: flex;
    width: 100%;
    flex-direction: column;
}

.company-content {
    display: flex;
    width: 100%;
    flex-direction: column;
}

.header {
    display: flex;
    width: 100%;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;

    &__avatar {
        width: 7em;
        margin-left: 100px;
        margin-top: 30px;
        margin-bottom: 30px;

    }

    &__info {
        margin-left: 20px;

        &__name {
            font-size: 2em;
            font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
            margin-bottom: 10px;
        }

        &__industry {
            font-size: 1.2em;
            font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
            margin-bottom: 10px;
            color: rgb(92, 92, 92);
        }
    }
}

.company-info {
    display: flex;
    width: 100%;
    height: 80vh;
    background-color: rgb(241, 245, 246);
    align-items: center;
    flex-direction: column;

    &__header-info {
        display: flex;
        width: 89%;
        flex-direction: row;
        margin-top: 30px;
        font-size: 1.3em;
        font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;

        &__item {
            display: flex;
            flex-direction: column;
            width: 33.33%;
            height: 7em;
            background-color: white;
            justify-content: center;
            border: 1px solid rgb(207, 207, 207);

            &__key {
                color: rgb(103, 103, 104);
                font-weight: 600;
                margin: 0 0 10px 30px;
            }

            &__value {
                margin: 10px 0 0 30px;
                color: rgb(26, 26, 26);
                font-weight: 700;
            }

            &--first {
                display: flex;
                flex-direction: column;
                width: 33.33%;
                background-color: white;
                justify-content: center;
                border: 1px solid rgb(207, 207, 207);
                border-radius: 7px 0 0 7px;
            }

            &--last {
                display: flex;
                flex-direction: column;
                width: 33.33%;
                background-color: white;
                justify-content: center;
                border: 1px solid rgb(207, 207, 207);
                border-radius: 0 7px 7px 0;
            }
        }
    }

    &__footer-info {
        display: flex;
        width: 89%;
        flex-direction: row;
        margin-top: 30px;

        &__contact {
            display: flex;
            flex-direction: column;
            width: 49%;
            background-color: white;
            border-radius: 10px;
            border: 1px solid rgb(207, 207, 207);

            &__title {
                font-size: 1.8em;
                font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
                margin: 0;
                width: 97.5%;
                padding: 15px 0 15px 20px;
                border-bottom: 1px solid rgb(207, 207, 207);

            }

            &__row {
                display: flex;
                width: 100%;
                flex-direction: row;
                margin-top: 20px;

                &--item {
                    margin-left: 50px;
                    margin-right: 200px;

                    &__key {
                        font-size: 1.5em;
                        font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
                        color: rgb(1, 43, 63);
                        margin: 0;
                    }

                    &__value {
                        font-size: 1.4em;
                        font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
                        color: rgb(4, 63, 90);
                        margin: 0;
                    }
                }
            }

            &__column {
                display: flex;
                width: 100%;
                flex-direction: column;
                margin: 20px 0 20px 50px;


                &__key {
                    font-size: 1.5em;
                    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
                    color: rgb(1, 43, 63);
                    margin: 0;
                }

                &__value {
                    font-size: 1.4em;
                    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
                    color: rgb(4, 63, 90);
                    margin: 0;
                }
            }
        }

        &__map {
            display: flex;
            width: 49%;
            margin-left: 2%;
            width: 100%;
            height: 238px;
            justify-content: center;
            align-items: center;
            background-color: white;
            border-radius: 10px;
            border: 1px solid rgb(207, 207, 207);
            font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;

            &__google-map {
                width: 100%;
                height: 238px;
                border-radius: 10px;

            }
        }
    }

}