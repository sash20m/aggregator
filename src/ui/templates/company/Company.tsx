/* eslint-disable camelcase */
import React, { useEffect, useState } from 'react';

import { GetServerSidePropsContext } from 'next';
import { useRouter } from 'next/router';
import { getCompany } from '../../../services/companies.services';
import { CompanyInfo } from '../../organisms/companyInfo/CompanyInfo';
import { HeaderBar } from '../../molecules/headerbar/Headerbar';

import './Company.scss';

interface Additional {
  lat: number;
  long: number;
}

interface AddressDeFacto {
  additional: Additional;
}

interface ContactInfo {
  emails: boolean[];
  phones: boolean[];
  address_de_facto: AddressDeFacto;
  sites: string[];
}

interface Size {
  name: string;
}

interface GeneralData {
  idno: string;
  creation_year: number;
  size: Size;
  contact_info: ContactInfo;
}

interface Status {
  title: string;
}

export interface Company {
  name: string;
  status: Status;
  general_data: GeneralData;
}

interface Props {
  data: Company;
}

export const CompanyPage: React.FC<Props> = (props) => {
  const { data } = props;
  const [imageURL, setImageURL] = useState('');
  const Router = useRouter();

  useEffect(() => {
    const route = Router.query;
    console.log(data, 'jheye');
    if (!data.name) {
      Router.push('/company/[slug]', `/company/${route.slug}`);
    } else if (data.general_data.contact_info.sites[0]) {
      setImageURL(new URL(data.general_data.contact_info.sites[0]).hostname);
    }
  }, [Router, data]);

  return (
    <div className="layout-company">
      <HeaderBar searchVisible />
      {data.name && <CompanyInfo companyData={{ data, imageURL }} />}
    </div>
  );
};

export const getSSProps = async ({
  params,
}: GetServerSidePropsContext): Promise<{
  props: {
    data: Company;
  };
} | null> => {
  if (params) {
    const data = await getCompany(params.slug);
    return {
      props: { data },
    };
  }
  return null;
};
