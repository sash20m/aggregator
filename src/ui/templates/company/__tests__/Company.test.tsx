import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import { CompanyPage } from '../Company';

const data = {
  name: 'Aridon',
  status: {
    title: 'Active',
  },
  general_data: {
    idno: '123456789',
    creation_date: 2020,
    size: {
      name: '10-20',
    },
    contact_info: {
      emails: [true],
      phones: [true, true],
      mobile: [true],
      address_de_facto: {
        additional: {
          lat: 20,
          long: 20,
        },
      },
      sites: ['www.aridon.com'],
    },
  },
};

afterEach(cleanup);

describe('company ', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<CompanyPage data={data} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be rendered', async () => {
    const { getByTestId } = render(<CompanyPage data={data} />);

    expect(getByTestId('company')).toBeInTheDocument();
  });

  it('should have 2 children', async () => {
    const { getByTestId } = render(<CompanyPage data={data} />);

    expect(getByTestId('company').childElementCount).toEqual(2);
  });
});
