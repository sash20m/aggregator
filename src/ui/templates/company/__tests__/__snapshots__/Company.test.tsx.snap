// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`company  should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="layout-company"
    data-testid="company"
  >
    <div
      class="header-main"
      data-testid="headerbar"
    >
      <div
        class="header-main__content"
      >
        <button
          class="logo__btn"
          type="button"
        >
          <img
            alt="Logo"
            class="logo"
            src="../logo-small.png"
          />
        </button>
        <div
          class="search"
        >
          <img
            alt=""
            class="search__icon"
            src="../search.png"
          />
          <input
            class="search__input"
            placeholder="Search from 225,195 companies"
            type="text"
            value=""
          />
          <button
            class="search__show-all"
            type="button"
          >
            Show All
          </button>
        </div>
      </div>
    </div>
    <div
      class="company-content"
      data-testid="company-content"
    >
      <div
        class="header"
      >
        <div
          class="header__content"
        >
          <div
            class="header__no-avatar"
          >
            A
          </div>
          <div
            class="header__info"
          >
            <div
              class="header__info__name"
              data-testid="company-name"
            >
              Aridon
            </div>
            <div
              class="header__info__industry"
            >
              Active
            </div>
          </div>
        </div>
      </div>
      <div
        class="company-info"
      >
        <div
          class="company-info__header-info"
        >
          <div
            class="company-info__header-info__item--first"
          >
            <p
              class="company-info__header-info__item__key"
            >
              IDNO
            </p>
            <p
              class="company-info__header-info__item__value"
            >
              123456789
            </p>
          </div>
          <div
            class="company-info__header-info__item"
          >
            <p
              class="company-info__header-info__item__key"
            >
              Registration year
            </p>
            <p
              class="company-info__header-info__item__value"
            >
              2020
            </p>
          </div>
          <div
            class="company-info__header-info__item--last"
          >
            <p
              class="company-info__header-info__item__key"
            >
              Staff
            </p>
            <p
              class="company-info__header-info__item__value"
            >
              10-20
            </p>
          </div>
        </div>
        <div
          class="company-info__footer-info"
        >
          <div
            class="company-info__footer-info__contact"
          >
            <p
              class="company-info__footer-info__contact__title"
            >
              Contact Information
            </p>
            <div
              class="company-info__footer-info__contact__row"
            >
              <div
                class="company-info__footer-info__contact__row--item"
              >
                <p
                  class="company-info__footer-info__contact__row--item__key"
                >
                  EMAIL:
                </p>
                <div
                  class="company-info__footer-info__contact__row--item__value"
                >
                  <img
                    alt="email"
                    class="company-info__footer-info__contact__row--item__value__icon"
                    src="../email.png"
                  />
                  <div
                    class="company-info__footer-info__contact__row--item__value__results"
                  >
                    <p>
                       Email
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="company-info__footer-info__contact__row--item"
              >
                <p
                  class="company-info__footer-info__contact__row--item__key"
                >
                  PHONE/FAX:
                </p>
                <div
                  class="company-info__footer-info__contact__row--item__value"
                >
                  <img
                    alt="phone"
                    class="company-info__footer-info__contact__row--item__value__icon"
                    src="../phone.png"
                  />
                  <div
                    class="company-info__footer-info__contact__row--item__value__results"
                    data-testid="phone"
                  >
                    <p>
                      Phone
                    </p>
                    <p>
                      Phone
                    </p>
                  </div>
                </div>
                <div
                  class="company-info__footer-info__contact__row--item__value"
                >
                  <img
                    alt="mobile"
                    class="company-info__footer-info__contact__row--item__value__icon"
                    src="../mobile.png"
                  />
                  <div
                    class="company-info__footer-info__contact__row--item__value__results"
                  >
                    <p>
                      Mobile
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="company-info__footer-info__contact__column"
            >
              <div
                class="company-info__footer-info__contact__column__key"
              >
                <img
                  alt="mobile"
                  class="company-info__footer-info__contact__row--item__value__icon"
                  src="../www.png"
                />
                Website
              </div>
              <p
                class="company-info__footer-info__contact__column__value"
              />
            </div>
          </div>
          <div
            class="company-info__footer-info__map"
          >
            <div />
            <div>
              Loading...
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
